test/src/unit-win-filesystem.cc

```diff
@@ -237,4 +243,17 @@ TEST_CASE_METHOD(WinFx, "Test Windows filesystem", "[windows]") {
   CHECK(win_.is_file(URI(test_file_path + "2").to_path()));
 }
 
+TEST_CASE_METHOD(
+    WinFx, "Test writing large files", "[.nightly_only][windows][large-file]") {
+  const uint64_t five_gigabytes = static_cast<uint64_t>(5) << 30;
+
+  std::string file = TEMP_DIR + "\\large-file";
+
+  std::vector<uint8_t> buffer(five_gigabytes);
+
+  REQUIRE(win_.write(file, buffer.data(), buffer.size()).ok());
+
+  REQUIRE(win_.read(file, 0, buffer.data(), buffer.size()).ok());
+}
+
```

Comment:
I wrote this test that creates a huge file and reads from it; we can't compare the data are the same because GitHub Actions runner images have only 7GB of RAM.

I still don't know how to make it run on the nightlies. I can't find of a way to pass arguments to `tiledb_unit` from CMake.

Reply:
for comparing data is same, more work, but pull in checksum code from somewhere and compute against data written and data read, compare at end, wouldn't require two buffers...

Reply:
Oh yeah, didn't think of it. 💡

Reply:
Eventually I want to add xxhash, but md5 is fine for now for this purpose. I guess technically this is uninitialized memory so we are "usually" checking "something" rather than zeros.

Reply:
> technically this is uninitialized memory

For me it looked like it was zero-initialized. I updated the test to write a pattern, and compare the MD5s of the buffers.

And I still have no idea how to make it run only on nightlies.

Reply:
I'm not sure what's more portable, but you could either add an environment variable to the nightly runner [here](https://github.com/TileDB-Inc/TileDB/blob/dev/.github/workflows/nightly-test.yml#L38-L39) or maybe add a bootstrap parameter `--is-nightly` that could be used to generate a compiler definition.